pyanaconda/payload/dnfpayload.py

```diff
@@ -381,35 +381,58 @@ def _add_repo(self, ksrepo):
         #     and use this new one.  The highest profile user of this is livecd
         #     kickstarts.
         if repo.id in self._base.repos:
-            if not url and not mirrorlist and not metalink:
-                self._base.repos[repo.id].enable()
-            else:
+            if url or mirrorlist or metalink:
                 with self._repos_lock:
                     self._base.repos.pop(repo.id)
                     self._base.repos.add(repo)
-                repo.enable()
         # If the repo's not already known, we've got to add it.
         else:
             with self._repos_lock:
                 self._base.repos.add(repo)
-            repo.enable()
 
-        # Load the metadata to verify that the repo is valid
+        log.debug("added repo: '%s' - %s", ksrepo.name, url or mirrorlist or metalink)
+
+    def _fetch_md(self, repo):
+        """Download repo metadata
+
+        :param repo: name/id of repo to fetch
+        :type repo: str
+        :returns: None
+        """
+        ksrepo = self._base.repos[repo]
+        ksrepo.enable()
         try:
-            self._base.repos[repo.id].load()
+            # Load the metadata to verify that the repo is valid
+            ksrepo.load()
         except dnf.exceptions.RepoError as e:
+            ksrepo.disable()
+            log.debug("repo: '%s' - %s failed to load repomd", ksrepo.name,
+                     ksrepo.baseurl or ksrepo.mirrorlist or ksrepo.metalink)
             raise payload.MetadataError(e)
 
-        log.info("added repo: '%s' - %s", ksrepo.name, url or mirrorlist or metalink)
+        log.info("enabled repo: '%s' - %s and got repomd", ksrepo.name,
+                 ksrepo.baseurl or ksrepo.mirrorlist or ksrepo.metalink)
 
     def addRepo(self, ksrepo):
-        """Add a repo to dnf and kickstart repo lists.
+        """Add an enabled repo to dnf and kickstart repo lists.
 
         :param ksrepo: Kickstart Repository to add
         :type ksrepo: Kickstart RepoData object.
         :returns: None
         """
         self._add_repo(ksrepo)
+        self._fetch_md(ksrepo.name)
+        super().addRepo(ksrepo)
+
+    def addDisabledRepo(self, ksrepo):
+        """Add a disabled repo to dnf and kickstart repo lists.
+
+        :param ksrepo: Kickstart Repository to add
+        :type ksrepo: Kickstart RepoData object.
+        :returns: None
+        """
+        ksrepo.disable()
+        self._add_repo(ksrepo)
```

Comment:
I think you should call `super().addDisabledRepo(ksrepo)` after this line. Otherwise parent will enable this repo back.

Reply:
good call!